File: Paradigmas de linguagem de programação/Lista de Exercicio - PLP/Questão 05/Python/main.py
class Animal:
    def som(self):
        raise NotImplementedError

class Vaca(Animal):
    def som(self):
        return "Vaca: Muuuu"

class Cabra(Animal):
    def som(self):
        return "Cabra: Bééééé"

def fazer_som_animais(animais):
    for animal in animais:
        print(animal.som())

if __name__ == "__main__":
    animais = [Vaca(), Cabra()]
    fazer_som_animais(animais)

File: Paradigmas de linguagem de programação/Lista de Exercicio - PLP/Questão 05/Java/Main.java
interface Animal {
    String som();
}

class Vaca implements Animal {
    public String som() {
        return "Vaca: Muuuu";
    }
}

class Cabra implements Animal {
    public String som() {
        return "Cabra: Béééé";
    }
}

public class Main {
    public static void fazerSomAnimais(Animal[] animais) {
        for (Animal animal : animais) {
            System.out.println(animal.som());
        }
    }

    public static void main(String[] args) {
        Animal[] animais = {new Vaca(), new Cabra()};
        fazerSomAnimais(animais);
    }
}
